src/ui/pages/AssetInfo/AssetHistory.tsx

```diff
@@ -0,0 +1,271 @@
+      />
+      <UnstyledButton onClick={handleDialogOpen} className={styles.historyItem}>
+        <div className={styles.historyItemBackdrop} />
+        <HStack
+          gap={12}
+          justifyContent="space-between"
+          style={{ position: 'relative', paddingBlock: 12 }}
+        >
+          <VStack gap={0} style={{ justifyItems: 'start' }}>
+            <UIText kind="body/regular">{actionTitle}</UIText>
+            <UIText kind="small/regular" color="var(--neutral-500)">
+              {actionDatetime}
+            </UIText>
+          </VStack>
+          <VStack gap={0} style={{ justifyItems: 'end' }}>
+            <UIText
+              kind="body/regular"
+              color={isIncoming ? 'var(--positive-500)' : 'currentColor'}
+            >
+              {actionBalance}
+            </UIText>
+            <UIText kind="small/regular" color="var(--neutral-500)">
+              {actionValue}
+            </UIText>
+          </VStack>
+        </HStack>
+      </UnstyledButton>
+      <CenteredDialog
+        ref={dialogRef}
+        containerStyle={{ backgroundColor: 'var(--neutral-100)' }}
+        renderWhenOpen={() => (
+          <>
+            <Button
+              kind="ghost"
+              value="cancel"
+              size={36}
+              style={{
+                width: 36,
+                padding: 8,
+                position: 'absolute',
+                top: 16,
+                left: 8,
+              }}
+              onClick={handleDialogDismiss}
+            >
+              <ArrowLeftIcon style={{ width: 20, height: 20 }} />
+            </Button>
+            <ActionDetailedView
+              action={action}
+              networks={networks}
+              address={address}
+            />
+          </>
+        )}
+      />
+    </>
+  );
+}
+
+export function AssetHistory({
+  assetId,
+  address,
+  assetFullInfo,
+}: {
+  assetId: string;
+  address: string;
+  assetFullInfo?: AssetFullInfo;
+}) {
+  const { networks } = useNetworks();
+  const { singleAddressNormalized, ready } = useAddressParams();
+  const { currency } = useCurrency();
+  const {
+    value,
+    // TODO: this flag doesn't work, needs to be fixed
+    isFetching: actionsAreLoading,
+    hasNext,
+    fetchMore,
+  } = useAddressActions(
+    {
+      address: singleAddressNormalized,
+      currency,
+      actions_fungible_ids: [assetId],
+    },
+    {
+      limit: 10,
+      listenForUpdates: true,
+      paginatedCacheMode: 'first-page',
+      enabled: ready,
+    }
+  );
+
+  const asset = assetFullInfo?.fungible;
+
+  if (actionsAreLoading || !asset || !networks || !value?.length) {
+    return null;
+  }
+
+  return (
+    <VStack gap={8} style={{ opacity: actionsAreLoading ? 0.8 : 1 }}>
+      <VStack gap={4}>
+        <UIText kind="headline/h3">History</UIText>
+        <VStack gap={0}>
+          {value.map((action) => (
+            <AssetHistoryItem
+              key={action.id}
```

Comment:
Sometimes in dev env I got this error
```
Encountered two children with the same key, `34fb69bd6e4d56b4841b013359984a72`
```
Probably because there exist duplicate action.id's

Reply:
changed to hash. Hopefully, this should be more reliable